### PR TITLE
Disabling invite a deactivated user

### PR DESCRIPTION
**Pull request by jingcheng16**
## Product Description

image

One of our customer thought, by invite an deactivated user who is already a member of the domain with a new role, the new role will overwrite the old role, which is not the case. Theoretically you should not invite someone who is already a member again. 
So I decide to disable invitation for deactivated user. This is discussed and approved in product meeting.

## Technical Summary

Ticket: https://dimagi.atlassian.net/browse/SAAS-15997

### Safety story

Tested locally. The change is very straightforward.

### QA Plan

No QA

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

File: corehq/apps/registration/forms.py [hunk 1]
```diff
@@ -592,6 +592,10 @@ def clean_email(self):
         if email.lower() in self.excluded_emails:
             raise forms.ValidationError(_("A user with this email address is already in "
                                           "this project or has a pending invitation."))
+        web_user = WebUser.get_by_username(email)
+        if web_user and not web_user.is_active:
+            raise forms.ValidationError(_("A user with this email address is deactivated. "
+                                          "Please reactivate this user first."))
```
Comment (orangejenny) [inline corehq/apps/registration/forms.py:598]: Not blocking: it'd be a nice UX improvement to turn "reactivate this user" into a link to the edit page for the relevant web user. (Granted, then you have to make sure whatever code catches this error ultimately displays it as HTML.)

**Review by orangejenny — APPROVED** [review]

File: corehq/apps/registration/forms.py (hunk 1 above)
Comment (jingcheng16) [reply corehq/apps/registration/forms.py:598]: Thanks Jenny. Usually we don't deactivate user, deactivation and reactivation can be operated by us in Admin>Look up user by email. So having that link might not be very helpful for most customers. For SSO user who get auto-deactivated, they need first add the user back to their identity provider, then ask the user to relogin, I feel this is too long in a error message?

Comment (orangejenny) [reply corehq/apps/registration/forms.py:598]: I see. If most users won't have access, perhaps we shouldn't tell them to reactivate the user? We have other messages that say to contact support (if that's the appropriate action), or maybe just removing the second sentence would be fine.

Comment (jingcheng16) [reply corehq/apps/registration/forms.py]: Yes, removed in 409c134ebfb67cf6f3fdd837c06f313146e526ee ...

**Review by orangejenny — APPROVED** [review]

**Pull request merged**